lib/rules/jquery-dollar-sign-reference.js

```diff
@@ -126,15 +126,34 @@ module.exports = {
       };
     }
 
+    function getValueNodes(node) {
+      var results = [];
+      if (node.type === 'LogicalExpression') {
+        results.push.apply(results, getValueNodes(node.left));
+        results.push.apply(results, getValueNodes(node.right));
+      } else if (node.type === 'ConditionalExpression') {
+        results.push.apply(results, getValueNodes(node.consequent));
+        results.push.apply(results, getValueNodes(node.alternate));
+      } else {
+        results.push(node);
+      }
+      return results;
+    }
+
     function checkForValidjQueryReference(node, left, right) {
       if (right == null) { return; }
       var isjQueryRef = isjQueryReference(left);
-      var isjQueryVal = isjQueryValue(right);
-      // console.log(right);
+      var valueNodeTypes = getValueNodes(right).map(isjQueryValue);
+      var hasDefinitejQueryValue = valueNodeTypes.some(function(nodeType) {
```

Comment:
I'd love to, but the [eslint plugin targets es5](https://github.com/Shopify/eslint-plugin-shopify/blob/master/package.json#L35), and doesn't have a transcompilation step.

I'm guessing this is because:
- We don't want to restrict our plugin to people using the latest & greatest node version
- We don't want to introduce a build step for a small project / just never had the time